core/domain/stats_jobs_continuous.py

```diff
@@ -411,27 +412,72 @@ def entity_classes_to_map_over(cls):
 
     @staticmethod
     def map(item):
+        # pylint: disable=line-too-long
         if InteractionAnswerSummariesMRJobManager._entity_created_before_job_queued(
                 item):
+            state_answers_dict = {
+                'exploration_id': item.exploration_id,
+                'exploration_version': item.exploration_version,
+                'state_name': item.state_name,
+                'interaction_id': item.interaction_id,
+                'submitted_answer_list': copy.deepcopy(
+                    item.submitted_answer_list)
+            }
 
-            # All visualizations desired for the interaction.
-            visualizations = interaction_registry.Registry.get_interaction_by_id(
-                item.interaction_id).answer_visualizations
-
-            # Get all desired calculations for the current state interaction id.
-            calc_ids = list(set(viz.calculation_id for viz in visualizations))
-            calculations = [
-                calculation_registry.Registry.get_calculation_by_id(calc_id)
-                for calc_id in calc_ids]
+            # Output answers submitted to the exploration for this exp version.
+            yield ('%s:%s:%s' % (
+                item.exploration_id, item.exploration_version,
+                item.state_name), state_answers_dict)
 
-            # Perform each calculation, and store the output.
-            for calc in calculations:
-                calc_output = calc.calculate_from_state_answers_entity(item)
-                calc_output.save()
+            # Output the same set of answers independent of the version. This
+            # allows the reduce step to aggregate answers across all
+            # exploration versions.
+            state_answers_dict['exploration_version'] = VERSION_ALL
```

Comment:
Do you need a deepcopy somewhere? Mildly concerned that you're modifying the dict which has previously been yielded.